File: Algorithms/Implementation/countingValleys.js
function countingValleys(n, s) {

    let l = 0,
        v = 0
  
    for (let i = n; i-- > 0;) {
      if (s.charAt(i) === 'U') {
        if (l++ == 0)
          ++v
      } else {
        --l
      }
    }
  
    return v
  }

File: Interview Preparation/Warm up/Python/countingValleys.py
# Complete the countingValleys function below.
def countingValleys(n, s):
    current_level = 0
    count = 0
    for i in range(n):
        if s[i] == 'U':
            current_level += 1
        elif s[i] == 'D':
            current_level -= 1
            if current_level == -1:
                count += 1
    return count
